incubator/group/msg.go

```diff
@@ -0,0 +1,59 @@
+package group
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+)
+
+var _ sdk.Msg = &MsgCreateGroup{}
+
+const (
+	msgTypeGroup = "create_group"
+)
+
+func (m MsgCreateGroup) Route() string { return ModuleName }
+func (m MsgCreateGroup) Type() string  { return msgTypeGroup }
+
+// GetSigners returns the addresses that must sign over msg.GetSignBytes()
+func (m MsgCreateGroup) GetSigners() []sdk.AccAddress {
+	// TODO: @aaronc ok with this constraint? We can enforce the signature on creation. Also see `MsgUpdateGroupAdmin.GetSigners()`.
+	return []sdk.AccAddress{m.Admin}
+}
+
+// GetSignBytes returns the bytes for the message signer to sign on
+func (m MsgCreateGroup) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
+}
+
+// ValidateBasic does a sanity check on the provided data
+func (m MsgCreateGroup) ValidateBasic() error {
+	if m.Admin.Empty() {
+		return sdkerrors.Wrap(ErrEmpty, "admin")
+	}
+	// TODO: @aaronc can we ensure that the group is never empty? That would simplify authZ logic later as we do not have
```

Comment:
I'm not sure we actually need to prevent this case. I can imagine scenarios where some custodial key management service creates a group before it actually has keys to populate it with. `admin` can't be empty so the group can be updated later (although I think it may be possible to set admin to nil eventually). Can you explain why this is necessary?

Reply:
An empty group is alright as long as the decision policy requires a minimum of 1 weight to pass. Otherwise there is not AuthZ. Rule of thump should be `Greater Than` in decision policies.
No need to set minimum number of members. What about a maximum count? At some point we would run into Gas limits but the error message would not be great.

Reply:
What do you mean by `AuthZ`?

I'd prefer not to have a maximum count. Large groups won't necessarily cause gas problems as long as they're not all added at once. Also even having a max count can't ensure gas won't be an issue.

Reply:
What do you mean by `AuthZ`?

I'd prefer not to have a maximum count. Large groups won't necessarily cause gas problems as long as they're not all added at once. Also even having a max count can't ensure gas won't be an issue.

Reply:
What do you mean by `AuthZ`?

I'd prefer not to have a maximum count. Large groups won't necessarily cause gas problems as long as they're not all added at once. Also even having a max count can't ensure gas won't be an issue.

Reply:
What do you mean by `AuthZ`?

I'd prefer not to have a maximum count. Large groups won't necessarily cause gas problems as long as they're not all added at once. Also even having a max count can't ensure gas won't be an issue.

Reply:
`AuthZ` stands for Authorization.  👌no max. There is no delete group msg specified so no worries about that scenario.